kafka/integration_test.go

```diff
@@ -1213,6 +1213,48 @@ func TestProducerConsumerHeaders(t *testing.T) {
 	c.Close()
 }
 
+// TestConsumerSeekPartitions tests seeking of partitions using SeekPartitions().
+func TestConsumerSeekPartitions(t *testing.T) {
+	if !testconfRead() {
+		t.Skipf("Missing testconf.json")
+	}
+
+	conf := ConfigMap{
+		"group.id": testconf.GroupID,
+	}
+
+	conf.updateFromTestconf()
+
+	consumer, err := NewConsumer(&conf)
+	if err != nil {
+		t.Fatalf("NewConsumer failed: %s", err)
+	}
+
+	tps := []TopicPartition{
+		{Topic: &testconf.Topic, Partition: 0},
+	}
+	err = consumer.Assign(tps)
+	if err != nil {
+		t.Fatalf("Assign failed: %s", err)
+	}
+
+	tps[0].Offset = OffsetBeginning
+	seekedPartitions, err := consumer.SeekPartitions(tps)
+	if err != nil {
+		t.Errorf("SeekPartitions failed: %s", err)
+	}
+	if len(seekedPartitions) != len(tps) {
+		t.Errorf(
+			"SeekPartitions should return result for %d partitions, %d returned",
+			len(tps), len(seekedPartitions))
+	}
+	for _, seekedPartition := range seekedPartitions {
+		if seekedPartition.Error != nil {
+			t.Errorf("Seek error for partition %v", seekedPartition)
+		}
+	}
+}
```

Comment:
Since this is an integration test, it should also check that it indeed receives messages starting at the seeked offset.
I recommend you produce 10 messages to a single partition, set auto.offset.reset=end on the consumer, assign(), and then call SeekPartitions() with offset 5, then do a ReadMessage and verify that the message's offset is indeed 5.

Reply:
Done